### PR TITLE
chore: point VERSION of source dev/builds to GH repo

### PR DESCRIPTION
**Pull request by caybro**
For these source/dev builds:
- export the "development" context variable
- display "git master SHA" as the version number
- disable the "Check for updates" button
- use APP_VERSION in nim

Rationale: instead of updating git master VERSION text file manually each
time an official version is released, we will now mark source/dev builds
with the corresponding git commit hash and display that in the
Settings/About section

### What does the PR do

Fixes version number for source/dev builds

### Affected areas

Settings/About

### Screenshot of functionality (including design for comparison)

- [x] I've checked the design and this PR matches it

Dev build of this PR:
image

AppImage produced from this PR:
image

### CODE REVIEW

File: src/nim_status_client.nim [hunk 1]
```diff
@@ -112,6 +112,7 @@ proc mainProc() =
   let appController = newAppController(statusFoundation)
   let networkAccessFactory = newQNetworkAccessManagerFactory(TMPDIR & "netcache")
 
+  let isDevelopmentQVariant = newQVariant(if defined(development): true else: false)
   let isProductionQVariant = newQVariant(if defined(production): true else: false)
```
Comment (jrainville) [inline src/nim_status_client.nim:116]: Can we just reuse `production` instead? I'm pretty sure they are and will stay mutually exclusive.

**Review by jrainville — COMMENTED** [review]
Looks good, I just wonder if we can use `producation` instead for the QML

File: src/nim_status_client.nim (hunk 1 above)
Comment (caybro) [reply src/nim_status_client.nim:116]: Yeah, you're right, will simplify this

Comment (caybro) [reply src/nim_status_client.nim]: @jrainville done

**Review by jrainville — APPROVED** [review]

**Review by jakubgs — APPROVED** [review]
I see, so the `VERSION` file only takes effect for releases, and normally we just get the commit.

Cool.

**Comment by caybro** [comment]
> I see, so the `VERSION` file only takes effect for releases, and normally we just get the commit.
> 
> Cool.

Yup, exactly

**Pull request merged**